ansq/tcp/connection.py

```diff
@@ -45,9 +45,16 @@ async def connect(self) -> bool:
         self.logger.debug(f"Connect to {self.endpoint} established")
 
         self._reader_task = self._loop.create_task(self._read_data_task())
+        self._reader_task.add_done_callback(self._reader_task_callback)
 
         return True
 
+    def _reader_task_callback(self, task: asyncio.Task) -> None:
+        exc = task.exception()
+
+        if exc is not None:
+            self._message_queue.close(exc=exc)
```

Comment:
```suggestion
            self._message_queue.put_nowait(None)
            self.logger.exception(exc)
```

I wonder, what if instead of bringing in the `CloseableQueue`, we close the generator by putting None into the message queue, which would close the message generator. What do you think?

Reply:
If there is more than one consumer?